src/Faker/Provider/Internet.php

```diff
@@ -120,7 +130,15 @@ public function domainWord()
     {
         $lastName = $this->generator->format('lastName');
 
-        return strtolower(static::transliterate($lastName));
+        $lastName = strtolower(static::transliterate($lastName));
+        $lastName = rtrim($lastName, '.');
+
+        // fallback in case transliterate() doesn't support the language and removed all letters
```

Comment:
Added a commit to throw exception instead. Still, this may be refined.

Please tell me before merging, I'd like to squash the commits a bit once we have settled on the end result.